Expand README and update for current state (#41)
geoarrow/geoarrow: README.md

# GeoArrow - geospatial data with Apache Arrow

> Specifications for storing geospatial data in Apache Arrow and Apache Parquet.

Reviewer: I'd probably remove `Parquet` from the first sentence, given that that might confuse people already familiar with GeoParquet. What do you think

The [Apache Arrow](https://arrow.apache.org/) project specifies a standardized
language-independent columnar memory format. It enables shared computational libraries,
zero-copy shared memory and streaming messaging, interprocess communication, etc and is
supported by many programming languages and data libraries.

Spatial information can be represented as a collection of discrete objects using points,
lines and polygons, i.e. vector data. The
[Simple Feature Access](https://www.ogc.org/standards/sfa) standard provides a widely
used abstraction, defining a set of geometries: Point, LineString, Polygon, MultiPoint,
MultiLineString, MultiPolygon, GeometryCollection. Next to a geometry, simple features
can also have non-spatial attributes that describe the feature.

> Geospatial data often comes in tabular format, with one (or multiple) column with

Reviewer: ```suggestion
Geospatial data often comes in tabular format, with one or more columns with
```

feature geometries and additional columns with feature attributes. The Arrow columnar
memory model is therefore perfectly suited to store such data, both vector features and

> its attribute data. The GeoArrow specification is about how the vector features

Reviewer: ```suggestion
their attribute data. The GeoArrow specification defines how the vector features
```

(geometries) can be stored in Arrow (and Arrow-compatible) data structures.

This repository contains the specifications for:

> - The memory layout for storing geometries natively in an Arrow field ([format.md](./format.md))

Reviewer: It might be nice to have a section that explicitly describes what we mean by "native". I understand native to mean O(1) coordinate access, so excluding WKB/WKT arrays.

Author: I think the format.md document now has explicit sections on native vs serialized (now Dewey's PR is merged). So maybe we should clarify it there if that's not yet the case?

(removed the "native" here, since the format.md also describes both)

- The Arrow extension type definitions ([extension-types.md](./extension-types.md))

Defining a standard and efficient way to store geospatial data in the Arrow memory
layout helps interoperability between different tools and ensures geospatial tools can
leverage the growing Apache Arrow ecosystem:

> - Efficient, columnar data formats. Leveraging the performant and compact storage of

Reviewer: I think using "file" here might help lessen confusion on arrow vs parquet
```suggestion
- Efficient, columnar file formats. Leveraging the performant and compact storage of
```

  Apache Parquet as a vector data format in geospatial tools using
  [GeoParquet](https://github.com/opengeospatial/geoparquet/).
- Accelerated between-process geospatial data exchange using Apache Arrow IPC message
  format and Apache Arrow Flight
- Zero-copy in-process geospatial data transport using the Apache Arrow C Data Interface
  (e.g., GDAL)
- Shared libraries for geospatial data type representation and computation for query
  engines that support columnar data formats (e.g., Velox, DuckDB, Acero)

The goal of this repository is to discuss and move towards specifications
on metadata and memory layout for storing geospatial data in the Arrow memory
layout or related formats.

> Relationship with GeoParquet: the GeoParquet specification originally started in this

Reviewer: I'd bold this or put it in an level ~3 heading?
